apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/logging/core/FileWriter.java

```diff
@@ -134,6 +136,41 @@ public Object call() throws Exception {
                     return null;
                 }
             });
+
+            if (Config.Logging.MAX_HISTORY_FILES > 0) {
```

Comment:
deletes expired logs when agent launch just only?

Reply:
no, this will be execute every time when log file rollover happen

Reply:
no, this will be execute every time when log file rollover happen

Reply:
i see. Thanks for your comment.